packages/catalog-client/src/CatalogClient.ts

```diff
@@ -91,6 +91,24 @@ export class CatalogClient implements CatalogApi {
     );
   }
 
+  // TODO: These need tests!
+  async getAttachment(name: EntityName, key: string): Promise<Blob> {
+    const url = await this.getAttachmentUrl(name, key);
+    const response = await fetch(url);
+
+    if (!response.ok) {
+      throw await ResponseError.fromResponse(response);
+    }
+
+    return await response.blob();
+  }
+
+  async getAttachmentUrl(name: EntityName, key: string): Promise<string> {
+    return `${await this.discoveryApi.getBaseUrl('catalog')}/entities/by-name/${
+      name.kind
+    }/${name.namespace}/${name.name}/attachments/${key}`;
```

Comment:
always `encodeURIComponent`, so we don't have to trust the input data :)

Reply:
Good idea, there are still some places in the client that don't do it. But that is something for another PR. I wonder whether we should validate the key of the attachment on emit, with similar rules as for entity names.

Reply:
Moved the remaining fixes to #5327